### ISSUE TITLE
Warning when opening project

### ISSUE THREAD

**Issue by kindlychung** [Groterik/qtcreator-dlangeditor#19]
Relevant parts of the .pro file:

```
## set the QTC_SOURCE environment variable to override the setting here
QTC_SOURCE = /Users/kaiyin/Qt/qt-creator
QTCREATOR_SOURCES = $$QTC_SOURCE
isEmpty(QTCREATOR_SOURCES):QTCREATOR_SOURCES=/usr/src/qtcreator
!exists($$QTCREATOR_SOURCES):\
    error("Set variable QTC_SOURCE to the QtCreator's sources path (current path is \"$$QTC_SOURCE\")")

## set the QTC_BUILD environment variable to override the setting here
QTC_BUILD = /Users/kaiyin/Qt/qt-creator-frameworks
IDE_BUILD_TREE = $$QTC_BUILD
isEmpty(IDE_BUILD_TREE):IDE_BUILD_TREE=/usr/lib/qtcreator
!exists($$IDE_BUILD_TREE): \
    error("Set variable QTC_BUILD to the QtCreator's libraries path (current path is \"$$QTC_BUILD\")")
```

I got an warning when I open the project in qt creator:

```
Project MESSAGE: Your QtCreator's sources version is 3.6.82
Project MESSAGE: Your QtCreator's sources major version is 3
Project MESSAGE: Your QtCreator's sources minor version is 6
Project MESSAGE: Plugin output path is /Users/kaiyin/Library/Application Support/QtProject/Qt Creator/plugins/3.6.82
Warnings while parsing QML type information of /Users/kaiyin/Qt/5.5/clang_64/qml:
<dump of /Users/kaiyin/Qt/5.5/clang_64/qml>:1:24: Reading only version 1.1 parts.
<dump of /Users/kaiyin/Qt/5.5/clang_64/qml>:10:5: Expected only Component and ModuleApi object definitions.
```

**Comment by kindlychung** [comment]
Got lots of errors when building:

```
18:31:06: Running steps for project dlangeditor...
18:31:06: Configuration unchanged, skipping qmake step.
18:31:06: Starting: "/usr/bin/make" 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -c -pipe -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -std=c++11 -stdlib=libc++ -mmacosx-version-min=10.7 -fvisibility=hidden -fvisibility-inlines-hidden -Wall -W -fPIC -DQTCREATOR_MAJOR_VERSION=3 -DQTCREATOR_MINOR_VERSION=6 -DQT_QML_DEBUG -DQT_PLUGIN -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_TESTLIB_LIB -DQT_CONCURRENT_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQT_TESTCASE_BUILDDIR='"/private/tmp/qtcreator-dlangeditor"' -Isrc -Ithirdparty/msgpack/include -I/Users/kaiyin/Qt/qt-creator-frameworks/src -I/Users/kaiyin/Qt/qt-creator/src/libs -I/Users/kaiyin/Qt/qt-creator/tools -I/Users/kaiyin/Qt/qt-creator/src/plugins -I/Users/kaiyin/Qt/qt-creator/src/libs/3rdparty -I. -I/Users/kaiyin/Qt/5.5/clang_64/lib/QtWidgets.framework/Headers -I/Users/kaiyin/Qt/5.5/clang_64/lib/QtGui.framework/Headers -I/Users/kaiyin/Qt/5.5/clang_64/lib/QtTest.framework/Headers -I/Users/kaiyin/Qt/5.5/clang_64/lib/QtConcurrent.framework/Headers -I/Users/kaiyin/Qt/5.5/clang_64/lib/QtNetwork.framework/Headers -I/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers -I.moc/debug-shared -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/AGL.framework/Headers -I/Users/kaiyin/Qt/5.5/clang_64/mkspecs/macx-clang -F/Users/kaiyin/Qt/5.5/clang_64/lib -o .obj/debug-shared/dlangeditorplugin.o src/dlangeditorplugin.cpp
In file included from src/dlangeditorplugin.cpp:1:
src/dlangeditorplugin.h:26:18: warning: 'aboutToShutdown' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    ShutdownFlag aboutToShutdown();
                 ^
/Users/kaiyin/Qt/qt-creator/src/libs/extensionsystem/iplugin.h:60:26: note: overridden virtual function is here
    virtual ShutdownFlag aboutToShutdown() { return SynchronousShutdown; }
                         ^
In file included from src/dlangeditorplugin.cpp:5:
src/codemodel/dcdmodel.h:51:10: warning: 'appendIncludePaths' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    void appendIncludePaths(const QStringList &includePaths);
         ^
src/codemodel/dmodel.h:102:18: note: overridden virtual function is here
    virtual void appendIncludePaths(const QStringList &includePaths) = 0;
                 ^
In file included from src/dlangeditorplugin.cpp:7:
src/codemodel/dastedmodel.h:52:10: warning: 'appendIncludePaths' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    void appendIncludePaths(const QStringList &includePaths);
         ^
src/codemodel/dmodel.h:102:18: note: overridden virtual function is here
    virtual void appendIncludePaths(const QStringList &includePaths) = 0;
                 ^
In file included from src/dlangeditorplugin.cpp:9:
src/dlangeditor.h:89:33: error: only virtual member functions can be marked 'override'
    QString defaultPath() const Q_DECL_OVERRIDE;
                                ^~~~~~~~~~~~~~~
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qcompilerdetection.h:1014:26: note: expanded from macro 'Q_DECL_OVERRIDE'
# define Q_DECL_OVERRIDE override
                         ^
In file included from src/dlangeditorplugin.cpp:9:
src/dlangeditor.h:90:39: error: only virtual member functions can be marked 'override'
    QString suggestedFileName() const Q_DECL_OVERRIDE;
                                      ^~~~~~~~~~~~~~~
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qcompilerdetection.h:1014:26: note: expanded from macro 'Q_DECL_OVERRIDE'
# define Q_DECL_OVERRIDE override
                         ^
src/dlangeditorplugin.cpp:131:53: error: calling 'projectInfo' with incomplete return type 'CppTools::ProjectInfo'
        CppTools::ProjectInfo pinfo = modelmanager->projectInfo(project);
                                      ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
/Users/kaiyin/Qt/qt-creator/src/plugins/cpptools/cppmodelmanager.h:96:17: note: 'projectInfo' declared here
    ProjectInfo projectInfo(ProjectExplorer::Project *project) const;
                ^
/Users/kaiyin/Qt/qt-creator/src/plugins/cpptools/cppmodelmanager.h:59:7: note: forward declaration of 'CppTools::ProjectInfo'
class ProjectInfo;
      ^
src/dlangeditorplugin.cpp:131:31: error: variable has incomplete type 'CppTools::ProjectInfo'
        CppTools::ProjectInfo pinfo = modelmanager->projectInfo(project);
                              ^
/Users/kaiyin/Qt/qt-creator/src/plugins/cpptools/cppmodelmanager.h:59:7: note: forward declaration of 'CppTools::ProjectInfo'
class ProjectInfo;
      ^
src/dlangeditorplugin.cpp:133:13: error: no type named 'type' in the global namespace
            foreach (const CppTools::ProjectPart::HeaderPath &header, pinfo.headerPaths()) {
            ^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:975:21: note: expanded from macro 'foreach'
#    define foreach Q_FOREACH
                    ^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:918:24: note: expanded from macro 'Q_FOREACH'
for (QForeachContainer<QT_FOREACH_DECLTYPE(container)> _container_((container)); \
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:903:87: note: expanded from macro 'QT_FOREACH_DECLTYPE'
#    define QT_FOREACH_DECLTYPE(x) typename QtPrivate::remove_reference<decltype(x)>::type
                                                                                    ~~^
src/dlangeditorplugin.cpp:133:13: error: C++ requires a type specifier for all declarations
            foreach (const CppTools::ProjectPart::HeaderPath &header, pinfo.headerPaths()) {
            ^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:975:21: note: expanded from macro 'foreach'
#    define foreach Q_FOREACH
                    ^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:918:56: note: expanded from macro 'Q_FOREACH'
for (QForeachContainer<QT_FOREACH_DECLTYPE(container)> _container_((container)); \
                                                       ^
src/dlangeditorplugin.cpp:133:51: error: no type named 'HeaderPath' in 'CppTools::ProjectPart'
            foreach (const CppTools::ProjectPart::HeaderPath &header, pinfo.headerPaths()) {
                           ~~~~~~~~~~~~~~~~~~~~~~~^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:921:10: note: expanded from macro 'Q_FOREACH'
    for (variable = *_container_.i; _container_.control; _container_.control = 0)
         ^
src/dlangeditorplugin.cpp:164:65: error: calling 'projectInfo' with incomplete return type 'CppTools::ProjectInfo'
                    CppTools::ProjectInfo pinfo = modelmanager->projectInfo(currentProject);
                                                  ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/kaiyin/Qt/qt-creator/src/plugins/cpptools/cppmodelmanager.h:96:17: note: 'projectInfo' declared here
    ProjectInfo projectInfo(ProjectExplorer::Project *project) const;
                ^
/Users/kaiyin/Qt/qt-creator/src/plugins/cpptools/cppmodelmanager.h:59:7: note: forward declaration of 'CppTools::ProjectInfo'
class ProjectInfo;
      ^
src/dlangeditorplugin.cpp:164:43: error: variable has incomplete type 'CppTools::ProjectInfo'
                    CppTools::ProjectInfo pinfo = modelmanager->projectInfo(currentProject);
                                          ^
/Users/kaiyin/Qt/qt-creator/src/plugins/cpptools/cppmodelmanager.h:59:7: note: forward declaration of 'CppTools::ProjectInfo'
class ProjectInfo;
      ^
src/dlangeditorplugin.cpp:166:25: error: no type named 'type' in the global namespace
                        foreach (const CppTools::ProjectPart::HeaderPath &header, pinfo.headerPaths()) {
                        ^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:975:21: note: expanded from macro 'foreach'
#    define foreach Q_FOREACH
                    ^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:918:24: note: expanded from macro 'Q_FOREACH'
for (QForeachContainer<QT_FOREACH_DECLTYPE(container)> _container_((container)); \
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:903:87: note: expanded from macro 'QT_FOREACH_DECLTYPE'
#    define QT_FOREACH_DECLTYPE(x) typename QtPrivate::remove_reference<decltype(x)>::type
                                                                                    ~~^
src/dlangeditorplugin.cpp:166:25: error: C++ requires a type specifier for all declarations
                        foreach (const CppTools::ProjectPart::HeaderPath &header, pinfo.headerPaths()) {
                        ^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:975:21: note: expanded from macro 'foreach'
#    define foreach Q_FOREACH
                    ^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:918:56: note: expanded from macro 'Q_FOREACH'
for (QForeachContainer<QT_FOREACH_DECLTYPE(container)> _container_((container)); \
                                                       ^
src/dlangeditorplugin.cpp:166:63: error: no type named 'HeaderPath' in 'CppTools::ProjectPart'
                        foreach (const CppTools::ProjectPart::HeaderPath &header, pinfo.headerPaths()) {
                                       ~~~~~~~~~~~~~~~~~~~~~~~^
/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers/qglobal.h:921:10: note: expanded from macro 'Q_FOREACH'
    for (variable = *_container_.i; _container_.control; _container_.control = 0)
         ^
3 warnings and 12 errors generated.
make: *** [.obj/debug-shared/dlangeditorplugin.o] Error 1
18:31:08: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project dlangeditor (kit: Desktop Qt 5.5.1 clang 64bit)
When executing step "Make"
18:31:08: Elapsed time: 00:02.
```

**Comment by Groterik** [comment]
That's happening because you are using the latest QtCreator sources to build the plugin with. Currently, I'm supporting QtC 3.6 only, since it's hard to follow all the changes in QtC's master branch.
If you are using QtCreator 3.6 (as a binary), you should build the plugin with the QtC 3.6 sources that can be downloaded from the [official page](http://download.qt.io/official_releases/qtcreator/3.6/3.6.0/).
